examples/lighting-app/lighting-common/gen/CHIPClustersObjc.mm

```diff
@@ -605,6 +605,279 @@ - (void)readAttributeClusterRevision:(ResponseHandler)completionHandler
 
 @end
 
+@interface CHIPNetworkCommissioning ()
+@property (readonly) Controller::NetworkCommissioningCluster cppCluster;
+@end
+
+@implementation CHIPNetworkCommissioning
+
+- (Controller::ClusterBase *)getCluster
+{
+    return &_cppCluster;
+}
+
+- (void)addThreadNetwork:(char *)operationalDataset
+              breadcrumb:(uint64_t)breadcrumb
+               timeoutMs:(uint32_t)timeoutMs
+       completionHandler:(ResponseHandler)completionHandler
+{
+    CHIPNetworkCommissioningClusterAddThreadNetworkResponseCallbackBridge * onSuccess
+        = new CHIPNetworkCommissioningClusterAddThreadNetworkResponseCallbackBridge(completionHandler, [self callbackQueue]);
+    if (!onSuccess) {
+        completionHandler([CHIPError errorForCHIPErrorCode:CHIP_ERROR_INCORRECT_STATE], nil);
+        return;
+    }
+
+    CHIPDefaultFailureCallbackBridge * onFailure = new CHIPDefaultFailureCallbackBridge(completionHandler, [self callbackQueue]);
+    if (!onFailure) {
+        delete onSuccess;
+        completionHandler([CHIPError errorForCHIPErrorCode:CHIP_ERROR_INCORRECT_STATE], nil);
+        return;
+    }
+
+    CHIP_ERROR err
+        = self.cppCluster.AddThreadNetwork(onSuccess->Cancel(), onFailure->Cancel(), operationalDataset, breadcrumb, timeoutMs);
+    if (err != CHIP_NO_ERROR) {
```

Comment:
in case of failure, are onSuccess/onFailure auto-cleared?

I am trying to understand why we only free memory on failure since it sseems like there is no delete on success.
Same for below.

Reply:
> in case of failure, are onSuccess/onFailure auto-cleared?
> 
> I am trying to understand why we only free memory on failure since it sseems like there is no delete on success.
> Same for below.

"in case of failure" == "in case of success" ? If so, the callback the callback is cleared once it has been fired (e.g  https://github.com/project-chip/connectedhomeip/blob/master/src/darwin/Framework/CHIP/gen/CHIPClustersObjc.mm#L50)

Now that I'm looking at it again I think one of the callback is leaking though since only one of them can fire. I will definitively fix it in an update to the API. Thanks !